### PR TITLE
FromBase64Transform.InputBlockSize set to 4 instead of 1

### PR DESCRIPTION
**Pull request by gfoidl**
Fixes https://github.com/dotnet/corefx/issues/39583 and sets `FromBase64Transform.CanTransformMultipleBlocks = true` (cf. https://github.com/dotnet/corefx/issues/39583#issuecomment-512854180).

### CODE REVIEW

File: src/System.Security.Cryptography.Encoding/src/System/Security/Cryptography/Base64Transforms.cs [hunk 1]
```diff
@@ -120,13 +120,13 @@ public FromBase64Transform(FromBase64TransformMode whitespaces)
...
-        private const int Base64InputBlockSize = 4;
         // A buffer with size 32 is stack allocated, to cover common cases and benefit from JIT's optimizations.
         private const int StackAllocSize = 32;
-        public int InputBlockSize => 1;
+
+        // Converting from Base64 generates 3 bytes output from each 4 bytes input block
+        public int InputBlockSize => 4;
         public int OutputBlockSize => 3;
-        public bool CanTransformMultipleBlocks => false;
+        public bool CanTransformMultipleBlocks => true;
```
Comment (gfoidl) [inline src/System.Security.Cryptography.Encoding/src/System/Security/Cryptography/Base64Transforms.cs]: > though that means it'll almost never use the stackalloc buffers
(from https://github.com/dotnet/corefx/issues/39583#issuecomment-512854180)

Are there any statistics about the typical `inputCount` (i.e. stream length on which CryptoStream should operate)? With these numbers the buffers could be tweaked (stackalloc / renting).

If not, I believe the current implementation of stackalloc / renting is still good.

Comment (bartonjs) [reply src/System.Security.Cryptography.Encoding/src/System/Security/Cryptography/Base64Transforms.cs]: > Are there any statistics about the typical inputCount (i.e. stream length on which CryptoStream should operate)? 

Nope.

The numbers I do have: apisof.net reports 1% APIPort usage for FromBase64Transform, and 0.4% for ToBase64Transform (25% CryptoStream)... and RC2 has almost more usage than that (somehow).  Convert.ToBase64String(byte[]) has 51.4% usage, and Convert.FromBase64String is 43,8% usage.

So "no idea what the inputs into this are, but it's definitely not the common way of doing Base64".

Comment (bartonjs) [inline src/System.Security.Cryptography.Encoding/src/System/Security/Cryptography/Base64Transforms.cs]: I'd like to see at least some test have to change as a result of this.  Can you add a test that just checks that the block sizes and can-transform match what we want them to be? :)

Comment (gfoidl) [reply src/System.Security.Cryptography.Encoding/src/System/Security/Cryptography/Base64Transforms.cs]: Tests are added with 886b05b77245f5f738e10a64eb118183855b8365

File: src/System.Security.Cryptography.Encoding/tests/Base64TransformsTests.cs [hunk 2]
```diff
@@ -253,5 +253,45 @@ public static void ValidateWhitespace(string expected, string data)
...
+                Assert.Equal(3, transform.OutputBlockSize);
+            }
+        }
+
+        [Fact]
+        public void TransformUsageFlags_ToBase64Transform()
+        {
+            using (var transform = new ToBase64Transform())
+            {
+                Assert.False(transform.CanTransformMultipleBlocks);
```
Comment (bartonjs) [inline src/System.Security.Cryptography.Encoding/tests/Base64TransformsTests.cs]: Is ToBase64Transform already effectively implemented as "true" here? (TransformBlock accepting an integer multiple of the input block size, TransformFinalBlock accepting any arbitrary amount of data) Or would flipping it to true actually cause failures?

Comment (gfoidl) [reply src/System.Security.Cryptography.Encoding/tests/Base64TransformsTests.cs]: > Is ToBase64Transform already effectively implemented as "true" here?

No it's not. It just transforms one block at a time.
https://github.com/dotnet/corefx/blob/886b05b77245f5f738e10a64eb118183855b8365/src/System.Security.Cryptography.Encoding/src/System/Security/Cryptography/Base64Transforms.cs#L37-L39
https://github.com/dotnet/corefx/blob/886b05b77245f5f738e10a64eb118183855b8365/src/System.Security.Cryptography.Encoding/src/System/Security/Cryptography/Base64Transforms.cs#L65-L71

See also https://github.com/dotnet/corefx/issues/39583#issuecomment-513304310 for this.

> Or would flipping it to true actually cause failures?

This test https://github.com/dotnet/corefx/blob/886b05b77245f5f738e10a64eb118183855b8365/src/System.Security.Cryptography.Encoding/tests/Base64TransformsTests.cs#L101 will fail with these https://github.com/dotnet/corefx/blob/886b05b77245f5f738e10a64eb118183855b8365/src/System.Security.Cryptography.Encoding/tests/Base64TransformsTests.cs#L22 test-data.

**Comment by maryamariyan** [comment]
Retriggering CI to see if Windows x86_Release still fails

**Comment by bartonjs** [comment]
`System.Net.Http.Functional.Tests.SocketsHttpHandlerTest_Http2.Http2_PendingReceive_SendsReset(doRead: False)` failed on Windows x64 Debug; but the details loader isn't opening.

Triggering again to hopefully get a clean run.

**Pull request merged**